data_dictionary/data_dictionary.cc

```diff
@@ -19,14 +19,163 @@
+keyspace::get_replication_strategy() const {
+    return _ops->get_replication_strategy(*this);
+}
+
+std::optional<keyspace>
+database::try_find_keyspace(std::string_view name) const {
+    return _ops->try_find_keyspace(*this, name);
+}
+
+bool
+database::has_keyspace(std::string_view name) const {
+    return bool(try_find_keyspace(name));
+}
+
+keyspace
+database::find_keyspace(std::string_view name) const {
+    auto ks = try_find_keyspace(name);
+    if (!ks) {
+        throw no_such_keyspace(name);
+    }
+    return *ks;
+}
+
+std::optional<table>
+database::try_find_table(std::string_view ks, std::string_view table) const {
+    return _ops->try_find_table(*this, ks, table);
+}
+
+table
+database::find_table(std::string_view ks, std::string_view table) const {
+    auto t = try_find_table(ks, table);
+    if (!t) {
+        throw no_such_column_family(ks, table);
+    }
+    return *t;
+}
+
+std::optional<table>
+database::try_find_table(utils::UUID id) const {
+    return _ops->try_find_table(*this, id);
+}
+
+table
+database::find_column_family(utils::UUID uuid) const {
+    auto t = try_find_table(uuid);
+    if (!t) {
+        throw no_such_column_family(uuid);
+    }
+    return *t;
+}
+
+schema_ptr
+database::find_schema(std::string_view ks, std::string_view table) const {
+    return find_table(ks, table).schema();
+}
+
+schema_ptr
+database::find_schema(utils::UUID uuid) const {
+    return find_column_family(uuid).schema();
+}
+
+bool
+database::has_schema(std::string_view ks_name, std::string_view cf_name) const {
+    return bool(try_find_table(ks_name, cf_name));
+}
+
+table
+database::find_column_family(schema_ptr s) const {
+    return find_column_family(s->id());
+}
+
+schema_ptr
+database::find_indexed_table(std::string_view ks_name, std::string_view index_name) const {
+    return _ops->find_indexed_table(*this, ks_name, index_name);
+}
+
+sstring
+database::get_available_index_name(std::string_view ks_name, std::string_view table_name,
+        std::optional<sstring> index_name_root) const {
+    return _ops->get_available_index_name(*this, ks_name, table_name, index_name_root);
+}
+
+std::set<sstring>
+database::existing_index_names(std::string_view ks_name, std::string_view cf_to_exclude) const {
+    return _ops->existing_index_names(*this, ks_name, cf_to_exclude);
+}
+
+schema_ptr
+database::get_cdc_base_table(sstring_view ks_name, std::string_view table_name) const {
+    return get_cdc_base_table(*find_table(ks_name, table_name).schema());
+}
+
+schema_ptr
+database::get_cdc_base_table(const schema& s) const {
+    return _ops->get_cdc_base_table(*this, s);
+}
+
+const db::extensions& 
+database::extensions() const {
+    return _ops->get_extensions(*this);
+}
+
+const gms::feature_service&
+database::features() const {
+    return _ops->get_features(*this);
+}
+
+const db::config&
+database::get_config() const {
+    return _ops->get_config(*this);
+}
+
+::database&
+database::real_database() const {
```

Comment:
Maybe you should add here a FIXME that this method should eventually be removed, as you explained in the commit message? A couple of years from now nobody will remember that the intention was to remove this method, and use it in more places :-(

Reply:
There is a comment in the declaration.